src/hash.ts

```diff
@@ -3,52 +3,130 @@ const createHash = require('create-hash')
-  if (typeof a === 'string' && !ethjsUtil.isHexString(a)) {
-    a = Buffer.from(a, 'utf8')
-  } else {
-    a = toBuffer(a)
-  }
-
-  if (!bits) bits = 256
-
+export const keccak = function(a: Buffer, bits: number = 256): Buffer {
+  assertIsBuffer(a)
   return createKeccakHash(`keccak${bits}`)
     .update(a)
     .digest()
 }
 
 /**
  * Creates Keccak-256 hash of the input, alias for keccak(a, 256).
- * @param a The input data (Buffer|Array|String|Number)
+ * @param a The input data (Buffer)
  */
-export const keccak256 = function(a: any): Buffer {
+export const keccak256 = function(a: Buffer): Buffer {
   return keccak(a)
 }
 
 /**
- * Creates SHA256 hash of the input.
- * @param a The input data (Buffer|Array|String|Number)
+ * Creates Keccak hash of a String input
+ * @param a The input data (String) If string is 0x-prefixed hex value it's
+ *          interpreted as hexadecimal, otherwise as utf8.
+ * @param bits (number = 256) The Keccak width
+ */
+export const keccakFromString = function(a: string, bits: number = 256) {
+  assertIsString(a)
+  let buf
+  if (!ethjsUtil.isHexString(a)) {
+    buf = Buffer.from(a, 'utf8')
+  } else {
+    buf = toBuffer(a)
+  }
+
+  return keccak(buf, bits)
+}
+
+/**
+ * Creates Keccak hash of a number array input
+ * @param a The input data (number[])
+ * @param bits (number = 256) The Keccak width
+ */
+export const keccakFromArray = function(a: number[], bits: number = 256) {
+  assertIsArray(a)
+  return keccak(toBuffer(a), bits)
+}
+
+/**
+ * Creates SHA256 hash of a Buffer input.
+ * @param a The input data (Buffer)
+ */
+export const sha256 = function(a: Buffer): Buffer {
+  assertIsBuffer(a)
+  return _sha256(a)
+}
+
+/**
+ * Creates SHA256 hash of a string input.
+ * @param a The input data (string)
+ */
+export const sha256FromString = function(a: string): Buffer {
+  assertIsString(a)
+  return _sha256(a)
+}
+
+/**
+ * Creates SHA256 hash of a number[] input.
+ * @param a The input data (number[])
  */
-export const sha256 = function(a: any): Buffer {
+export const sha256FromArray = function(a: number[]): Buffer {
+  assertIsArray(a)
+  return _sha256(a)
+}
+
+/**
+ * Creates SHA256 hash of an input.
+ * @param  a The input data (Buffer|Array|String)
+ */
+const _sha256 = function(a: any): Buffer {
   a = toBuffer(a)
   return createHash('sha256')
     .update(a)
     .digest()
 }
 
+/**
+ * Creates RIPEMD160 hash of a Buffer input.
+ * @param a The input data (Buffer)
+ * @param padded Whether it should be padded to 256 bits or not
+ */
+export const ripemd160 = function(a: Buffer, padded: boolean): Buffer {
+  assertIsBuffer(a)
+  return _ripemd160(a, padded)
+}
+
+/**
+ * Creates RIPEMD160 hash of a string input.
+ * @param a The input data (String)
+ * @param padded Whether it should be padded to 256 bits or not
+ */
+export const ripemd160FromString = function(a: string, padded: boolean): Buffer {
+  assertIsString(a)
+  return _ripemd160(a, padded)
+}
```

Comment:
I have a tendency not to introduce the hex string version methods for `sha256` and `ripemd160` since both are not so widely used, did some org-wide search here as well (not sure if this is a valid argument).

If you feel otherwise (that hex methods should be introduced) please do so.

Reply:
That sounds good to me - really just did this for the sake of symmetry. Will update.